tests/Unit/SystemCtlTest.php

```diff
@@ -3,27 +3,188 @@
+use SystemCtl\Command\CommandInterface;
+use SystemCtl\Exception\UnitNotFoundException;
 use SystemCtl\SystemCtl;
 use SystemCtl\Unit\Service;
 use SystemCtl\Unit\Timer;
 
+/**
+ * Class SystemCtlTest
+ *
+ * @package SystemCtl\Test\Unit
+ */
 class SystemCtlTest extends TestCase
 {
-    public function testGetServiceWithName()
+    /**
+     * @return ObjectProphecy
+     */
+    private function buildCommandDispatcherStub(): ObjectProphecy
     {
-        $systemctl = new SystemCtl();
+        $commandDispatcherStub = $this->prophesize(CommandDispatcherInterface::class);
+        $commandDispatcherStub->setTimeout(Argument::type('int'))->willReturn($commandDispatcherStub);
+        $commandDispatcherStub->setBinary(Argument::type('string'))->willReturn($commandDispatcherStub);
 
-        $service = $systemctl->getService('testService');
+        return $commandDispatcherStub;
+    }
+
+    /**
+     * @param string $output
+     *
+     * @return ObjectProphecy
+     */
+    private function buildCommandStub(string $output): ObjectProphecy
+    {
+        $command = $this->prophesize(CommandInterface::class);
+        $command->getOutput()->willReturn($output);
+
+        return $command;
+    }
+
+    /**
+     * @test
+     */
+    public function itShouldCallCommandDispatcherWithListUnitsAndUnitPrefixOnServiceGetting()
+    {
+        $unitName = 'testService';
+        $output = ' testService.service     Active running';
+        $commandDispatcherStub = $this->buildCommandDispatcherStub();
+        $commandDispatcherStub
+            ->dispatch(...['list-units', $unitName . '*'])
+            ->willReturn($this->buildCommandStub($output));
+
+        $systemctl = (new SystemCtl())->setCommandDispatcher($commandDispatcherStub->reveal());
+
+        $service = $systemctl->getService($unitName);
+        $this->assertInstanceOf(Service::class, $service);
+        $this->assertEquals('testService', $service->getName());
+    }
+
+    /**
+     * @test
+     */
+    public function itShouldReturnAServiceOnServiceGetting()
+    {
+        $unitName = 'testService';
+        $output = ' testService.service     Active running';
+        $commandDispatcherStub = $this->buildCommandDispatcherStub();
+        $commandDispatcherStub
+            ->dispatch(...['list-units', $unitName . '*'])
+            ->willReturn($this->buildCommandStub($output));
+
+        $systemctl = (new SystemCtl())->setCommandDispatcher($commandDispatcherStub->reveal());
+
+        $service = $systemctl->getService($unitName);
         $this->assertInstanceOf(Service::class, $service);
+    }
+
+    /**
+     * @test
+     */
+    public function itShouldReturnAServiceWithTheCorrectNameOnServiceGetting()
+    {
+        $unitName = 'testService';
+        $output = ' testService.service     Active running';
+        $commandDispatcherStub = $this->buildCommandDispatcherStub();
+        $commandDispatcherStub
+            ->dispatch(...['list-units', $unitName . '*'])
```

Comment:
Here you could simply do `->dispatch('list-units', $unitName . '*')`
On the other tests as well.